memory/src/main/scala/filodb.memory/format/vectors/IntBinaryVector.scala

```diff
@@ -274,34 +296,138 @@ trait IntVectorDataReader extends VectorDataReader {
 
 object OffheapSignedIntVector32 extends IntVectorDataReader {
   final def apply(vector: BinaryVectorPtr, n: Int): Int = UnsafeUtils.getInt(vector + 8 + n * 4)
+  final def sum(vector: BinaryVectorPtr, start: Int, end: Int): Long = {
+    var addr = vector + 8 + start * 4
+    val untilAddr = vector + 8 + end * 4 + 4   // one past the end
+    var sum: Long = 0L
+    while (addr < untilAddr) {
+      sum += UnsafeUtils.getInt(addr)
+      addr += 4
+    }
+    sum
+  }
 }
 
 object OffheapSignedIntVector16 extends IntVectorDataReader {
   final def apply(vector: BinaryVectorPtr, n: Int): Int = UnsafeUtils.getShort(vector + 8 + n * 2).toInt
+  final def sum(vector: BinaryVectorPtr, start: Int, end: Int): Long = {
+    var addr = vector + 8 + start * 2
+    val untilAddr = vector + 8 + end * 2 + 2   // one past the end
+    var sum = 0L
+    while (addr < untilAddr) {
+      sum += UnsafeUtils.getShort(addr)
+      addr += 2
+    }
+    sum
+  }
 }
 
 object OffheapSignedIntVector8 extends IntVectorDataReader {
   final def apply(vector: BinaryVectorPtr, n: Int): Int = UnsafeUtils.getByte(vector + 8 + n).toInt
+  final def sum(vector: BinaryVectorPtr, start: Int, end: Int): Long = {
+    var addr = vector + 8 + start
+    val untilAddr = vector + 8 + end + 1     // one past the end
+    var sum = 0L
+    while (addr < untilAddr) {
+      sum += UnsafeUtils.getByte(addr)
+      addr += 1
+    }
+    sum
+  }
 }
 
 object OffheapUnsignedIntVector16 extends IntVectorDataReader {
   final def apply(vector: BinaryVectorPtr, n: Int): Int =
     (UnsafeUtils.getShort(vector + 8 + n * 2) & 0x0ffff).toInt
+  final def sum(vector: BinaryVectorPtr, start: Int, end: Int): Long = {
+    val startRoundedUp = (start + 3) & ~3
+    var sum = defaultSum(vector, start, Math.min(end, startRoundedUp - 1))
```

Comment:
How will end-of-time-series marker be done for a Long column type data ? With Double valued columns, NaN is used currently.

Reply:
There are multiple ways we can handle end of time series marker, but the general solution is out of scope for this PR.  Since only Prometheus clients send NaN for end of time series marker, and only for doubles, let's implement a solution to skip/detect NaNs for DoubleColumns only.